experimental/LieAlgebras/src/LieAlgebraModule.jl

```diff
@@ -1006,18 +1030,28 @@ function exterior_power(V::LieAlgebraModule{C}, k::Int) where {C<:FieldElem}
     return Tuple(basis(V, i) for i in inds)
   end
 
+  mult_map = MapFromFunc(
+    Hecke.TupleParent(Tuple([zero(V) for f in 1:k])), E, my_mult, my_decomp
+  )
+  inv_mult_map = MapFromFunc(E, domain(mult_map), my_decomp, my_mult)
+
   set_attribute!(
     E,
-    :type => :exterior_power,
-    :power => k,
-    :base_module => V,
-    :exterior_pure_function => pure,
-    :exterior_pure_preimage_function => inv_pure,
+    :is_exterior_power => (V, k),
+    :wedge_pure_function => mult_map,
+    :wedge_generator_decompose_function => inv_mult_map,
     :embedding_tensor_power => T,
     :embedding_tensor_power_embedding => E_to_T,
     :embedding_tensor_power_projection => T_to_E,
   )
-  return E
+
+  cached && (_exterior_powers(V)[k] = (E, mult_map))
+
+  return E, mult_map
+end
+
+@attr Dict{Int,Tuple{typeof(V),MapFromFunc}} function _exterior_powers(V::LieAlgebraModule)
```

Comment:
```suggestion
@attr Dict{Int, <:Tuple{typeof(V), <:MapFromFunc}} function _exterior_powers(V::LieAlgebraModule)
```
I think in the constructor of dicts you can use abstract types. But for type assertion, I always needed the `<:`. Did this code even work before?

Reply:
I played around with it. Having abstract types in the dict constructor makes type inference nearly impossible. Changing the type assertion didn't change anything for me. But intuitively, I would say, that the type assertion should exactly match the type of the dict created here. So I would keep it as it currently is.